clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -538,12 +535,11 @@ public void subscribe(final Collection<String> topics) {
     public void unsubscribe() {
         acquireAndEnsureOpen();
         try {
-            Timer timer = time.timer(Long.MAX_VALUE);
-            ShareUnsubscribeEvent unsubscribeApplicationEvent = new ShareUnsubscribeEvent(calculateDeadlineMs(timer));
-            applicationEventHandler.add(unsubscribeApplicationEvent);
             log.info("Unsubscribing all topics");
 
-            subscriptions.unsubscribe();
+            Timer timer = time.timer(Long.MAX_VALUE);
```

Comment:
Should we replace `Long.MAX_VALUE` by `defaultApiTimeoutMs`?

Reply:
Yes, I think that's better.